cli/cmd/pro.go

```diff
@@ -92,10 +103,190 @@ func executeRemoteUpdateToken(ctx context.Context, client *kube.Client, namespac
+	if err != nil {
+		return nil, fmt.Errorf("cannot get latest offsets: %s", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("cannot get latest offsets: %d", resp.StatusCode)
+	}
+
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read response body: %s", err)
+	}
+	return data, nil
+}
+
+func updateOdigletDaemonSet(ctx context.Context, client *kube.Client, ns string, revert bool) *appsv1.DaemonSet {
+	ds, err := client.Clientset.AppsV1().DaemonSets(ns).Get(ctx, k8sconsts.OdigletDaemonSetName, metav1.GetOptions{})
+	if err != nil {
+		fmt.Println(fmt.Sprintf("\033[31mERROR\033[0m Unable to get Odiglet DaemonSet: %s", err))
+		os.Exit(1)
+	}
+
+	// Add offsets volume (if not already exist)
+	volumes := ds.Spec.Template.Spec.Volumes
+	if volumes == nil {
+		volumes = make([]v1.Volume, 0)
+	}
+	addVolume := true
+	for _, vol := range volumes {
+		if vol.Name == k8sconsts.GoOffsetsConfigMap {
+			addVolume = false
+			break
+		}
+	}
+	if addVolume {
+		volumes = append(volumes, v1.Volume{Name: k8sconsts.GoOffsetsConfigMap,
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: k8sconsts.GoOffsetsConfigMap,
+					},
+				},
+			},
+		})
+		ds.Spec.Template.Spec.Volumes = volumes
+	}
+
+	// Add offsets volume mounnt (if not already exist)
+	volumeMounts := ds.Spec.Template.Spec.Containers[0].VolumeMounts
+	if volumeMounts == nil {
+		volumeMounts = make([]v1.VolumeMount, 0)
+	}
+	addVolumeMount := true
+	for _, vm := range volumeMounts {
+		if vm.Name == k8sconsts.GoOffsetsConfigMap {
+			addVolumeMount = false
+			break
+		}
+	}
+	if addVolumeMount {
+		volumeMounts = append(volumeMounts, v1.VolumeMount{Name: k8sconsts.GoOffsetsConfigMap, MountPath: k8sconsts.OffsetFileMountPath})
+		ds.Spec.Template.Spec.Containers[0].VolumeMounts = volumeMounts
+	}
+
+	// Add offsets Env Var (if not already exist)
+	envVars := ds.Spec.Template.Spec.Containers[0].Env
+	if envVars == nil {
+		envVars = make([]v1.EnvVar, 0)
+	}
+	addEnvVar := true
+	for _, env := range envVars {
+		if env.Name == k8sconsts.GoOffsetsEnvVar {
+			addEnvVar = false
+			break
+		}
+	}
+	if addEnvVar {
+		envVars = append(envVars, v1.EnvVar{Name: k8sconsts.GoOffsetsEnvVar, Value: k8sconsts.OffsetFileMountPath + "/" + k8sconsts.GoOffsetsFileName})
+		ds.Spec.Template.Spec.Containers[0].Env = envVars
+	}
+	return ds
+}
+
 func init() {
 	rootCmd.AddCommand(proCmd)
 
 	proCmd.Flags().String("onprem-token", "", "On-prem token for Odigos")
 	proCmd.MarkFlagRequired("onprem-token")
 	proCmd.PersistentFlags().BoolVarP(&updateRemoteFlag, "remote", "r", false, "use odigos ui service in the cluster to update the onprem token")
+
+	proCmd.AddCommand(offsetsCmd)
+	offsetsCmd.Flags().StringVarP(&namespaceFlag, "namespace", "n", consts.DefaultOdigosNamespace, "target k8s namespace for Odigos installation")
```

Comment:
as mentioned in another comment, I support auto detecting ns like other commands.
If we choose to keep it, update the cmd text (copy-paste from install) :)